src/Listeners/UpdateSplitTitleAfterDiscussionWasRenamed.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/*
+* This file is part of flagrow/flarum-ext-split.
+*
+* Copyright (c) Flagrow.
+*
+* http://flagrow.github.io
+*
+* For the full copyright and license information, please view the license.md
+* file that was distributed with this source code.
+*/
+
+namespace Flagrow\Split\Listeners;
+
+use Illuminate\Contracts\Events\Dispatcher;
+use Flarum\Event\DiscussionWasRenamed;
+
+use Flarum\Core\Repository\PostRepository;
+
+class UpdateSplitTitleAfterDiscussionWasRenamed {
+    /**
+     * @param Dispatcher $events
+     */
+    public function subscribe(Dispatcher $events)
+    {
+        $events->listen(DiscussionWasRenamed::class, [$this, 'whenDiscussionWasRenamed']);
+    }
+
+    /**
+     * @param \Flarum\Event\DiscussionWasRenamed $event
+     */
+    public function whenDiscussionWasRenamed(DiscussionWasRenamed $event) {
+            
+        $this->posts = app(PostRepository::class);
+        $allPosts = $this->posts->
+                query()->
+                where('type',"=","discussionSplit")->
+                where('content',"like","%".$event->discussion->id."-%")->
```

Comment:
I think this is indeed very odd behaviour. You already said yourself. Let's see.


`%".$event->discussion->id."-%"` will look for all discussion split event posts that hold that content. What we should be doing is get the full URL to the discussion and remove the slug. That would give us a far more exact match.